datajoint/table.py

```diff
@@ -116,29 +116,55 @@ def get_select_fields(self, select_fields=None):
         """
         return '*' if select_fields is None else self.heading.project(select_fields).as_sql
 
-    def parents(self, primary=None):
+    def parents(self, primary=None, as_objects=False):
         """
         :param primary: if None, then all parents are returned. If True, then only foreign keys composed of
             primary key attributes are considered.  If False, the only foreign keys including at least one non-primary
             attribute are considered.
-        :return: dict of tables referenced with self's foreign keys
+        :param as_objects: if False (default), the output is a dict describing the foreign keys. If True, return table objects.
+        :return: dict of tables referenced with self's foreign keys  or list of table objects if as_objects=True
         """
-        return self.connection.dependencies.parents(self.full_table_name, primary)
+        parents = self.connection.dependencies.parents(self.full_table_name, primary)
+        if as_objects:
+            parents = [FreeTable(self.connection, c) for c in parents]
+        return parents
 
-    def children(self, primary=None):
+    def children(self, primary=None, as_objects=False):
         """
         :param primary: if None, then all children are returned. If True, then only foreign keys composed of
             primary key attributes are considered.  If False, the only foreign keys including at least one non-primary
             attribute are considered.
-        :return: dict of tables with foreign keys referencing self
-        """
-        return self.connection.dependencies.children(self.full_table_name, primary)
-
-    def descendants(self):
-        return self.connection.dependencies.descendants(self.full_table_name)
-
-    def ancestors(self):
-        return self.connection.dependencies.ancestors(self.full_table_name)
+        :param as_objects: if False (default), the output is a dict describing the foreign keys. If True, return table objects.
+        :return: dict of tables with foreign keys referencing self or list of table objects if as_objects=True
+        """
+        nodes = dict((next(iter(self.connection.dependencies.children(k).items())) if k.isdigit() else (k, v))
+                     for k, v in self.connection.dependencies.children(self.full_table_name, primary).items())
+        if as_objects:
+            nodes = [FreeTable(self.connection, c) for c in nodes]
+        return nodes
+
+    def descendants(self, as_objects=False):
+        nodes = [node for node in self.connection.dependencies.descendants(self.full_table_name) if not node.isdigit()]
```

Comment:
Also unclear why this is a subset now than prior implementation.

Reply:
prior implementation included the the alias nodes that represent renamed foreign keys. Now it only returns the nodes that represent tables.